Blog: Statsig vs PostHog (#7308)
PostHog/posthog.com: contents/blog/posthog-vs-statsig.mdx

---
title: "In-depth: PostHog vs Statsig"

> date: 2023-12-06

Author: ```suggestion
date: 2023-12-14
```

- **Statsig** offers advanced statistical methods for running tests. It lets you choose between Bayesian and Frequentist engines, run multi-armed bandit tests, and supports Bonferroni correction. It's ideal for data scientists and growth teams who need to conduct complex A/B testing.

- **PostHog** is ideal for teams who prefer a more straightforward approach to experimentation. It's an all-in-one platform built to help engineers understand users and create better products. Beyond [A/B testing](/ab-testing) and [feature flags](/feature-flags), it offers advanced [product analytics](/docs/product-analytics), [session replays](/docs/session-replay), [surveys](/docs/surveys), and more.

This post compares their platforms, experimentation features, pricing, and more.

## How is PostHog different?

### 1. It is an all-in-one platform

PostHog brings together all the tools engineers need for testing, releasing, and measuring the success of new features. Feature flags and A/B testing are only part of a suite of tools PostHog offers.

![PostHog homepage features](../images/blog/posthog-vs-statsig/all.png)

> PostHog combines usage, performance, and behavioral data with flags and experiments. PostHog’s [data warehouse](/docs/data-warehouse) also enables you to pull in data from external sources.
> 
> Having all these product and data tools together enables you to do better analysis of shipped features and make better decisions about what you are building next. 

Author: ```suggestion
PostHog combines usage, performance, and behavioral data with flags and experiments. Having all these product and data tools together enables you to do better analysis of shipped features and make better decisions about what you are building next. 
```

### 2. PostHog is open source and transparent

PostHog is built with transparency at its core. Not only do we work in the open and give full access to [our source code](https://github.com/PostHog/posthog), we also enable others to [build integrations](/docs/apps/build) or [other services on top of PostHog](/blog/how-we-built-an-app-server), open their own PRs, or give feedback on [our roadmap](/roadmap). PostHog's open app framework makes it easy to integrate internal tools, an advantage closed-source products like Statsig can't offer.

### 3. PostHog constantly ships new features

At PostHog, we ship weirdly fast. We update [our changelog](https://posthog.com/blog/posthog-changelog) with a recap of new features every week, and often there’s even more in beta testing. And that’s without counting all the apps and integrations submitted by our community! We work hard to keep PostHog on the cutting edge and we’re transparent about the ways we do that.

> 

Author: Would love feedback on this. They do also publish a [changelog](https://statsig.com/updates), but given that they're also dev-focused, I wasn't sure what else to put.

---
title: "In-depth: PostHog vs Statsig"
date: 2023-12-06
author: ["lior-neu-ner"]
showTitle: true
rootpage: /blog
sidebar: Blog
hideAnchor: true
featuredImage: ../images/blog/posthog-engineering-blog.png
featuredImageType: full
category: General
tags:
    - Comparisons
---

import { ComparisonTable } from 'components/ComparisonTable'
import { ComparisonHeader } from 'components/ComparisonTable/header'
import { ComparisonRow } from 'components/ComparisonTable/row'

PostHog and Statsig both offer A/B testing and feature flags, but they're different in two important ways:

> - **Statsig** offers advance experimentation tools, such as multi-armed bandit and holdout testing. It's ideal for data scientists, product managers, and growth teams who need to conduct complex A/B testing.

Reviewer: Statsig being warehouse native is a big different

Author: Mentioned in the table below, but I dont call this out here since I think most people are looking primarily at a/b testing comparision (it's their main feature)

> - **PostHog** is an all-in-one platform built to help engineers understand users and create better products. Beyond [feature flags](/feature-flags) and [A/B testing](/ab-testing), it includes [product analytics](/docs/product-analytics), [session replay](/docs/session-replay), [surveys](/docs/surveys), and more.

Reviewer: Also statsig includes some analytics, but PostHog's is much more advanced.

> - **PostHog** is an all-in-one platform built to help engineers understand users and create better products. Beyond [feature flags](/feature-flags) and [A/B testing](/ab-testing), it includes [product analytics](/docs/product-analytics), [session replay](/docs/session-replay), [surveys](/docs/surveys), and more.

Author: Reworked.

> large enterprises

They support startups too, so won't focus on that (They are self-serve + have an affordable plan)

This post compares their platforms, experimentation features, pricing, and more.

## How is PostHog different?

### 1. It is an all-in-one platform

PostHog brings together all the tools engineers need for testing, releasing, and measuring the success of new features. Feature flags and A/B testing are only part of a suite of tools PostHog offers.

![PostHog homepage features](../images/blog/posthog-vs-statsig/all.png)

> PostHog combines usage, performance, and behavioral data with flags and experiments. PostHog’s [data warehouse](/docs/data-warehouse) also enables you to pull in data from external sources.

Reviewer: I don't know if I'd mention data warehouse yet, we don't really know how well it will work with A/B testing and feature flags.

Author: @EDsCODE any thoughts?

Author: @ivanagas FYI this is on the growthbook blog https://posthog.com/blog/posthog-vs-growthbook

Reviewer 2: Hmm not sure where my other comment went. must've not pressed send earlier. 

To be safe, I would agree with @ivanagas and avoid drawing connections between data warehouse and experimentation for now. We're still in an experimental/validation phase with connecting external data

Having all these product and data tools together enables you to do better analysis of shipped features and make better decisions about what you are building next. 

> ### 2. PostHog is built for engineers

Reviewer: +1

Anyone can use PostHog – just [ask our marketing team](/blog/posthog-marketing) – but PostHog is designed to meet the needs of engineers. Our goal is to help developers [be better at product](/blog/helping-engineers-to-product).

### 3. We’re transparent (in many ways)

PostHog is open source. Our code, culture, and strategy are public [on GitHub](https://github.com/PostHog/posthog) and our [handbook](/handbook). While Statsig open sourced some code (like SDKs), most is closed source.

We're also self-serve. No need to "talk to sales" or "schedule a demo" unless you want to – we're always [happy to chat](/book-a-demo). You can sign up and try all the features PostHog offers for free.

## Comparing PostHog and LaunchDarkly

> ### Platform

Reviewer 3: Should add notebooks to this given it'll be live for everyone soon.

<ComparisonTable column1="PostHog" column2="Statsig">
  <ComparisonRow column1={true} column2={true} feature="A/B testing" description="Test changes and analyze impact" />
  <ComparisonRow column1={true} column2={true} feature="Feature flags" description="Roll out features safely; toggle features for cohorts or individuals" />
  <ComparisonRow column1={true} column2={true} feature="Product analytics" description="Track events and conversion; analyze user behavior" />

>   <ComparisonRow column1={false} column2={true} feature="Configurator" description="Replace hard-coded values in your app with config values" />

Reviewer: ```suggestion
  <ComparisonRow column1={false} column2={true} feature="Dynamic config" description="Replace hard-coded values in your app with config values" />
```

  <ComparisonRow column1={true} column2={false} feature="Session replays" description="Watch real users use your product; debug behavior" />
  <ComparisonRow column1={true} column2={false} feature="User surveys" description="Ask users for qualitative feedback and gather responses" />
  <ComparisonRow column1={true} column2={false} feature="Open source" description="Code publicly accessible" />

>   <ComparisonRow column1="In beta" column2={false} feature="Data warehouse storage" description="Store large amounts of analytics data" />

Reviewer: You can use a data warehouse with statsig? https://statsig.com/warehouse

Author: They offer the ability to run A/B tests on data within your warehouse, but they don't offer the actual storage of it. I tried to make this clear with these two points, but let me know if you think it can be clearer

>   <ComparisonRow column1={false} column2={true} feature="Data warehouse experiments" description="Run A/B tests natively in your existing warehouse" />

Reviewer 2: Yeah I would say so! Maybe a NIT is `running A/B tests on the data within your existing data warehouse`. The current phrasing tripped me out for a second because I just wasn't sure what running an A/B test in a warehouse meant

</ComparisonTable>

### Experimentation

> Experimentation is where PostHog and Statsig's functionality diverges. Both enable you to create [A/B/n](/tutorials/abn-testing) and [multivariate](/product-engineers/what-is-multivariate-testing-examples) tests, set custom goals, and calculate statistical significance, but:
> 
> - **Statsig** provides more options for statistical depth, such as Bayesian and Frequentist engines, and Bonferroni correction. It also offers advanced testing techniques, like multi-armed bandit, mutually exclusive, and holdout tests. This makes it well-suited for teams who are expect to run dozens of experiments on the same surface concurrently.
> 
> - **PostHog** provides a more intuitive user interface and simpler setup process, making it ideal for teams who prefer a more straightforward approach to experimentation.

Reviewer 3: ```suggestion
Both tools enable you to run [A/B/n](/tutorials/abn-testing) and [multivariate](/product-engineers/what-is-multivariate-testing-examples) tests, set custom goals, and calculate statistical significance, but:

- **Statsig** offers some more advanced testing techniques, such as multi-armed bandit, mutually exclusive, and holdout tests. It also lets you choose between Bayesian and Frequentist engines, and supports Bonferroni correction.

- **PostHog** provides a more intuitive user interface and simpler setup process, making it ideal for teams who prefer a more straightforward approach to experimentation, and tight integration with its other powerful tools.
```

  <ComparisonRow column1={true} column2={false} feature="Payloads" description="Flags with string, number, or JSON payloads" />
  <ComparisonRow column1={true} column2={true} feature="Percentage rollouts" description="Target percentages of a group" />
  <ComparisonRow column1={true} column2={true} feature="Custom targeting" description="Target users based on user properties, custom contexts" />
  <ComparisonRow column1={false} column2={true} feature="Scheduling" description="Schedule flags to turn on or off" />
  <ComparisonRow column1="Partial" column2={true} feature="Environments" description="Manage flags for dev, stage, prod" />
  <ComparisonRow column1={true} column2={true} feature="Bootstrapping" description="Flags available on frontend application load" />
  <ComparisonRow column1={true} column2={false} feature="Early access management" description="Manage betas, test features" />
</ComparisonTable>

### Product analytics

Both PostHog and Statsig offer the functionality you expect from product analytics tools, such as dashboards, graphs, and funnels. However:

> - **PostHog** provides deep insights into how users are interacting with your product. It includes features such as lifecycle, stickiness, correlation, and retention analysis.

Reviewer: SQL?

Author: Called out in the table. I think its fine not to mention it in this line as it breaks the flow of the sentence a bit

- **Statsig** does provide some additional insights, such as retention analysis, but its main focus is to use product analytics to set up metrics to run A/B tests on. 

<ComparisonTable column1="PostHog" column2="Statsig">
  <ComparisonRow column1={true} column2={false} feature="Autocapture" description="Capture events without manual logging" />
  <ComparisonRow column1={true} column2={true} feature="Dashboards" description="Combine insights into shareable dashboards" />
  <ComparisonRow column1={true} column2={true} feature="Graphs and trends" description="Build custom insights and visualizations" />
  <ComparisonRow column1={true} column2={true} feature="Cohorts" description="Combine users based on properties and events for group analysis" />

>   <ComparisonRow column1={true} column2={true} feature="Group analytics" description="Track metrics at a company level" />

Reviewer: ```suggestion
  <ComparisonRow column1={true} column2={false} feature="Group analytics" description="Track metrics at a company level" />
```

Reviewer: No mention of group or account-level analytics I can find.

Author: They do, they call it CustomID though. 

https://docs.statsig.com/client/concepts/user
https://docs.statsig.com/messages/serverRequiredUserID
https://docs.statsig.com/guides/experiment-on-custom-id-types

  <ComparisonRow column1={true} column2={true} feature="Funnels" description="Track users through a sequence of events" />
  <ComparisonRow column1={true} column2={true} feature="Retention analysis" description="Visualize which users stay, for how long" />
  <ComparisonRow column1={true} column2={false} feature="User paths" description="Track user flows and where they drop-off" />
  <ComparisonRow column1={true} column2={false} feature="Correlation analysis" description="Suggested events and properties that lead to success or failure" />
  <ComparisonRow column1={true} column2={false} feature="Lifecycle analysis" description="Understand who is dormant, churning, and thriving" />
  <ComparisonRow column1={true} column2={false} feature="Stickiness insights" description="See how many times users perform an event in a period of time." />
  <ComparisonRow column1={true} column2={false} feature="Formulas" description="Use custom formulas to calculate unique insights" />
  <ComparisonRow column1={true} column2={false} feature="Query editor" description="Write your own queries in SQL" />
</ComparisonTable>

> ### Pricing

Reviewer 3: Interested in @ivanagas's take here, but I'm inclined to ditch this entire pricing section. It's hard to parse and just comes out as us being more expensive for reasons that aren't super clear.

Even if we keep the testing bit, I'd definitely drop the analytics bit as it's an apples to oranges comparison.

Reviewer: - Agree about analytics.
- I think the section (especially the PostHog part) could be simplified with less details about the pricing per event tiers. Maybe a paragraph each and the image.
- Remove example scenarios.

Author: - Removed example scenarios section
- Removed analytics pricing section, although added a call out with the link to its pricing page (I do think its important to mention, since we call out that product analytics as a differentiating factor in this blog)
- Reworked text slightly

Both PostHog and Statsig offer simple, transparent pricing.

> **Statsig** has a free plan for up to 1 million *metered events* (a metered event is any feature flag, A/B testing, or logging event). Thereafter, it's $150 per month for up to 5 million metered events, and then $50 per every 1 million events. 

Reviewer: warehouse native also has its own plan.

Author: I think it's fine to not mention it. This blog doesn't focus too much on warehouse comparison, and I suspect that companies want warehouse native will expect to pay extra for it.

            <td><strong>PostHog cost</strong></td>
            <td><strong>Statsig cost</strong></td>
        </tr>
    </thead>
    <tbody>
       <tr>
            <td>1,000,000</td>
            <td>$0</td>
            <td>$0</td>
        </tr>
       <tr>
            <td>5,000,000</td>
            <td>$700</td>
            <td>$150</td>
        </tr>
       <tr>
            <td>10,000,000</td>
            <td>$1,350</td>
            <td>$400</td>
        </tr>
       <tr>
            <td>20,000,000</td>
            <td>$2,400</td>
            <td>$900</td>
        </tr>
    </tbody>
</table>
</div>

### Integrations

Both PostHog and Statsig have a range of integrations that enable them to import, export, enhance, and make use of data. However, a key difference is that since PostHog is open source, it's possible to [create your own integration](/docs/apps/build).

Below is a sample comparison of PostHog and Statsig's integrations. Be sure to checkout [PostHog's full list of integrations](/cdp).

<ComparisonTable column1="PostHog" column2="Statsig">
   <ComparisonRow column1={true} column2={true} feature="Imports" description="Import data from source" />

>    <ComparisonRow column1={true} column2={true} feature="Exports" description="Export data to other sources" />

Reviewer: These are both enterprise features, should we mention that?

Author: `Data Warehouse Imports` is an enterprise feature, but `Incoming Data Integrations`  and `Outgoing Data Integrations` are available on their Pro plan
